File: 10/js/2.js
const fs = require('fs');
const { create_cpu, create_crt } = require("./CPULib");

(function () {
  const contents = fs.readFileSync("../input.txt").toString();

  const ops = contents.split("\r\n").filter(x => x.length > 0).map(x => x.split(" "));

  const cpu = create_cpu(),
    crt = create_crt(40, 3);
  cpu.setCRT(crt);

  ops.forEach(op => cpu.exec(op[0], op.slice(1)));

  console.log(`x = ${cpu.getX()} after ${cpu.getCycle()} cycles`);
  console.log("===== [ CRT ] =====");
  console.log(crt.get());
})();

File: 10/hs/2.hs
import System.IO
import CPULib

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let
    ops = map ((\(i:a) -> (i, a)).words) $ lines contents
    cpu = create_cpu 1
    rc = create_record False 0 0
    crt = create_crt True 40 3
    (cpu', _, crt') = cpu_exec_all (cpu, rc, crt) ops
  print $ "x = " ++ (show $ fst cpu') ++ " after " ++ (show $ snd cpu') ++ " cycles."
  putStrLn "===== [ CRT ] ====="
  putStrLn $ crt_get crt'
  hClose h